kolibri/core/assets/src/state/actions.js

```diff
@@ -223,6 +223,36 @@ function getCurrentSession(store) {
   .catch(error => { handleApiError(store, error); });
 }
 
+function getFacilityConfig(store) {
+  const coreApp = require('kolibri');
+  const FacilityCollection = coreApp.resources.FacilityResource
+    .getCollection()
+    .fetch();
+
+  return FacilityCollection.then(facilities => {
+    const currentFacilityId = facilities[0].id;
+    const facilityConfigCollection = coreApp.resources.FacilityDatasetResource
+      .getCollection({ facility_id: currentFacilityId })
+      .fetch();
+    return facilityConfigCollection
```

Comment:
I always like reducing layers of indirection when possible, and I'd love to more clearly define what each layer is for - and if there isn't a good reason, remove it.

Just for additional context we've generally tried to follow this pattern:

* resource layer should basically mirror server state
* vuex state and getters should basically mirror the app-wide UI
* view computed properties should perform component-specific transformations

Very often the visible UI basically does mirror the server state, leading to the tedious mappers. In some situations though, it might not - e.g. if the UI only needs a subset of the API's data (using `fields` would help with this) or requires multiple requests or subsequent transformation.

At one point we'd even considered tying the resource layer more closely to vuex state, and trying to reference server models directly through vuex getters. However it seemed it harder to write code that performed efficient server queries and deal with asynchronous loading logic...